altsrc/flag.go

```diff
@@ -121,28 +133,34 @@ func (f *IntSliceFlag) ApplyInputSourceValue(cCtx *cli.Context, isc InputSourceC
 
 // ApplyInputSourceValue applies a Bool value to the flagSet if required
 func (f *BoolFlag) ApplyInputSourceValue(cCtx *cli.Context, isc InputSourceContext) error {
-	if f.set != nil && !cCtx.IsSet(f.Name) && !isEnvVarSet(f.EnvVars) && isc.isSet(f.BoolFlag.Name) {
-		value, err := isc.Bool(f.BoolFlag.Name)
-		if err != nil {
-			return err
-		}
-		for _, name := range f.Names() {
-			_ = f.set.Set(name, strconv.FormatBool(value))
+	if f.set != nil && !cCtx.IsSet(f.Name) && !isEnvVarSet(f.EnvVars) {
```

Comment:
@dearchap @james-prysm Do you think there is too mush nesting? Should we return early when don't need to handle. For example:
```
	if f.set == nil || cCtx.IsSet(f.Name) || isEnvVarSet(f.EnvVars) {
		return nil
	}
	for _, name := range f.BoolFlag.Names() {
		if !isc.isSet(name) {
			continue
		}
		value, err := isc.Bool(name)
		if err != nil {
			return err
		}
		for _, n := range f.Names() {
			_ = f.set.Set(n, strconv.FormatBool(value))
		}
	}
```

Reply:
@Dokiys definitely your approach is more readable. You want to make the changes and push ?